src/Elastica/Client.php

```diff
@@ -48,8 +48,19 @@ public function request($path, $method = Request::GET, $data = [], array $query
         }
 
         $response = parent::request($path, $method, $data, $query, $contentType);
-        $responseData = $response->getData();
 
+        $transportInfo = $response->getTransferInfo();
+
+        $connection = $this->getLastRequest()->getConnection();
+        $forbiddenHttpCodes = $connection->hasConfig('http_error_codes') ? $connection->getConfig('http_error_codes') : [];
+
+        if (isset($transportInfo['http_code']) && in_array($transportInfo['http_code'], $forbiddenHttpCodes, true)) {
+            $body = isset($transportInfo['body']) ? $transportInfo['body'] : 'blank';
+            $message = sprintf('Error in transportInfo: response code is %s, response body is %s', $transportInfo['http_code'], $body);
+            throw new \Exception($message);
```

Comment:
For debugging reasons we should throw a different exception. Maybe one of https://github.com/ruflin/Elastica/tree/master/lib/Elastica/Exception ? @ruflin ?

Reply:
SGTM. The part I'm worried here is that this could be a breaking change in Elastica? Want to open a PR against Elastica to discuss it there?